server/src/workspace/workspace-manager/workspace-manager.service.ts

```diff
@@ -200,23 +211,193 @@ export class WorkspaceManagerService {
 
   /**
    *
-   * Reset all standard objects and fields metadata for a given workspace
+   * Sync all standard objects and fields metadata for a given workspace and data source
+   * This will update the metadata if it has changed and generate migrations based on the diff.
    *
    * @param dataSourceId
    * @param workspaceId
    */
-  public async resetStandardObjectsAndFieldsMetadata(
+  public async syncStandardObjectsAndFieldsMetadata(
     dataSourceId: string,
     workspaceId: string,
   ) {
-    await this.objectMetadataService.deleteMany({
-      workspaceId: { eq: workspaceId },
+    const standardObjects = MetadataParser.parseAllMetadata(
+      [WebhookRecord, ApiKeyRecord, ViewSortRecord],
+      workspaceId,
+      dataSourceId,
+    );
+    const objectsInDB = await this.objectMetadataRepository.find({
+      where: { workspaceId, dataSourceId, isCustom: false },
+      relations: ['fields'],
     });
 
-    await this.createStandardObjectsAndFieldsMetadata(
-      dataSourceId,
-      workspaceId,
+    // We want to convert the objects and fields to a map to be able to compare them more easily
+    const mapByName = (arr: ObjectMetadataEntity[]) => {
+      return arr.reduce((acc, curr) => {
+        acc[curr.nameSingular] = {
+          ...curr,
+          fields: curr.fields.reduce((acc, curr) => {
+            acc[curr.name] = curr;
+            return acc;
+          }, {}),
+        };
+        return acc;
+      }, {});
+    };
+
+    const objectsInDBByName = mapByName(objectsInDB);
+    const standardObjectsByName = mapByName(standardObjects);
+
+    const objectPropertiesToIgnore = [
```

Comment:
I think we should also include them in the comparaison in case we change the way these fields or object properties behave